e2e-test-server/src/main/java/com/google/cloud/opentelemetry/endtoend/Server.java

```diff
@@ -102,11 +105,20 @@ private void respond(final String testId, final Response response) {
           public void onSuccess(String messageId) {}
 
           public void onFailure(Throwable t) {
-            System.out.println("failed to publish response to test: " + testId);
             t.printStackTrace();
           }
         },
         MoreExecutors.directExecutor());
+    try {
+      // Wait for the future to get completed.
+      // This prevents cloud functions from exiting too quickly
+      messageIdFuture.get(30, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+      Thread.currentThread().interrupt();
```

Comment:
Why is this necessary? (The interrupt)

I don't think this is really doing anything here.  If you want to `rethrow` feel free, but unless you think another bit of code *will* block in this thread, setting the interrupt flag isn't really doing anything AFAICT

Reply:
Yes, I do not see a reason for interrupting the thread again on an interrupt. I should be able to remove it and add a log statement there instead - indicating a timeout after 30 seconds.